src/main/java/com/car/app/controller/CarDao.java

```diff
@@ -39,10 +46,23 @@
 	 * getting all cars
 	 * @return List of cars   list of cars
 	 */
-	public List<Car> findAllCars() {
-		List<Car> resultList = entityManager.createQuery("select c from Car c").getResultList();
+	public PaginationObject<CarDto> findAllCars(int size, int page, String sort, String order, String search, String searchBy) {
+		sort = sort != null && !sort.trim().isEmpty() ? sort : "asc";
+		order = order != null && !order.trim().isEmpty() ? order : "id";
+		size = size == 0 ? 10 : size;
+		String query = filteringAndSortingQuery(sort, order, search, searchBy);
+		Query dr = entityManager.createQuery(query);
+		
+		// get total elements
+		int totalElmenets = dr.getResultList().size();
+		dr.setFirstResult(page * size);
+		dr.setMaxResults(size);
+		List<Car> resultList = dr.getResultList();
```

Comment:
This is the second time you're executing a query, when the amount of data is a few hundred or a few thousand of rows, there is not performance issues, but when the amount of data grows to hundred of thousand or million of rows then firing two queries like this might have an impact in the performance. Could you please try to reduce the amount of queries ? Sometimes is not possible, but we should always keep performance of database queries in mind.